net/httpclient.go

```diff
@@ -2,18 +2,219 @@ package net
 
 import (
 	"crypto/tls"
+	"io"
 	"net/http"
 	"net/http/httptrace"
+	"net/url"
+	"strings"
 	"time"
 
 	"github.com/opentracing/opentracing-go"
 	"github.com/opentracing/opentracing-go/ext"
+	"github.com/sirupsen/logrus"
+	"github.com/zalando/skipper/logging"
+	"github.com/zalando/skipper/secrets"
 )
 
 const (
 	defaultIdleConnTimeout = 30 * time.Second
+	defaultRefreshInterval = 5 * time.Minute
 )
 
+// StaticSecret implements SecretsReader interface. Example:
+//
+//    sec := []byte("mysecret")
+//    sss := StaticSecret(sec)
+//    b,_ := sss.GetSecret("")
+//    string(b) == sec // true
+type StaticSecret []byte
+
+// GetSecret returns the static secret
+func (st StaticSecret) GetSecret(string) ([]byte, bool) {
+	return st, true
+}
+
+// Close implements secrets.SecretsReader.
+func (st StaticSecret) Close() {}
+
+// StaticDelegateSecret delegates with a static string to the wrapped
+// SecretsReader
+type StaticDelegateSecret struct {
+	sr  secrets.SecretsReader
+	key string
+}
+
+// NewStaticDelegateSecret creates a wrapped secrets.SecretsReader,
+// that use given s to the underlying secrets.SecretsReader to return
```

Comment:
s -> key to be more descriptive.

Reply:
I am not sure if this StaticDelegateSecret is useful, tbh, because using a fixed key is not a complicated thing for the user code.

Reply:
StaticDelegateSecret can be used with kubernetes secrets, if you only have one, that you use for all client calls, for example.

Reply:
fair enough